src/ansible_builder/_target_scripts/introspect.py

```diff
@@ -344,7 +344,9 @@ def sanitize_requirements(collection_py_reqs):
                 consolidated.append(req)
                 seen_pkgs.add(req.name)
         except Exception as e:
-            logger.warning('Warning: failed to parse requirements from %s, error: %s', collection, e)
+            msg = f'Failed to parse requirements from {collection}, error: {e}'
+            logger.error(msg)
+            sys.exit(msg)
```

Comment:
This causes the error message to be printed twice:

```
[2/3] STEP 11/12: RUN $PYCMD /output/scripts/introspect.py introspect --sanitize --user-pip=requirements.txt --write-bindep=/tmp/src/bindep.txt --write-pip=/tmp/src/requirements.txt
Failed to parse requirements from user, error: Expected end or semicolon (after name and no valid version specifier)
    six=0.1
       ^
Failed to parse requirements from user, error: Expected end or semicolon (after name and no valid version specifier)
    six=0.1
       ^
```

Probably enough to just `sys.exit(1)` here.